packages/core/content-manager/admin/src/components/ConfigurationForm/Fields.tsx

```diff
@@ -33,6 +39,130 @@ import type { EditLayout } from '../../hooks/useDocumentLayout';
 type FormField = ConfigurationFormData['layout'][number]['children'][number];
 type Field = Omit<ConfigurationFormData['layout'][number]['children'][number], '__temp_key__'>;
 
+/* -------------------------------------------------------------------------------------------------
+ * Drag and Drop
+ * -----------------------------------------------------------------------------------------------*/
+
+const DroppableContainer = ({ id, children }: { id: string; children: React.ReactNode }) => {
+  const { setNodeRef } = useDroppable({
+    id,
+  });
+
+  return <div ref={setNodeRef}>{children}</div>;
+};
+
+export const SortableItem = ({ id, children }: { id: string; children: React.ReactNode }) => {
+  const { attributes, setNodeRef, transform, transition } = useSortable({
+    id,
+  });
+
+  const style = {
+    transform: CSS.Transform.toString(transform),
+    transition,
+  };
+
+  return (
+    <div ref={setNodeRef} style={style} {...attributes}>
+      {children}
+    </div>
+  );
+};
+
+const useDragAndDrop = (layout: ConfigurationFormData['layout']) => {
```

Comment:
ℹ️ The hook isn't really necessary I just found it helpful to isolate as much of the dnd logic as possible